src/mount.ts

```diff
@@ -20,16 +26,80 @@ export type MountOptions = {
   prepareContainer?: (container: HTMLElement) => void;
 };
 
+export type EnzymePropSelector = {
+  [key: string]: any;
+};
+
+export type ComponentType<Props> =
+  | ComponentClass<Props>
+  | FunctionComponent<Props>;
+
+export type EnzymeSelector = string | ComponentType<any> | EnzymePropSelector;
+
+/**
+ * Inspired by DefinitelyTyped's ReactWrapper.
+ * @see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cebb88fecfa52f854826e216a537867c11b0150e/types/enzyme/index.d.ts
+ */
+export type ComponentWrapper<P = {}, S = {}, C = Component> = {
+  at(index: number): ComponentWrapper;
+
+  childAt(index: number): ComponentWrapper<any, any>;
+  childAt<P2, S2>(index: number): ComponentWrapper<P2, S2>;
+
+  closest<P2>(
+    statelessComponent: FunctionComponent<P2>,
+  ): ComponentWrapper<P2, never>;
+  closest<P2>(component: ComponentType<P2>): ComponentWrapper<P2, any>;
+  closest(props: EnzymePropSelector): ComponentWrapper<any, any>;
+  closest(selector: string): ComponentWrapper<JSX.HTMLAttributes, any>;
+
+  exists(selector?: EnzymeSelector): boolean;
+
+  find<P2>(
+    statelessComponent: FunctionComponent<P2>,
+  ): ComponentWrapper<P2, never>;
+  find<P2>(component: ComponentType<P2>): ComponentWrapper<P2, any>;
+  find<C2 extends Component>(
+    componentClass: ComponentClass<C2['props']>,
+  ): ComponentWrapper<C2['props'], C2['state'], C2>;
+  find(props: EnzymePropSelector): ComponentWrapper<any, any>;
+  find(selector: string): ComponentWrapper<JSX.HTMLAttributes, any>;
+
+  first(): ComponentWrapper<P, S, C>;
+  forEach(
+    fn: (wrapper: ComponentWrapper<P, S, C>, index: number) => any,
+  ): ComponentWrapper<P, S, C>;
+  getDOMNode(): HTMLElement;
+  hasClass(className: string | RegExp): boolean;
+  instance(): C;
```

Comment:
From a quick look in the client it seems like `instance()` is only used on DOM elements, where `getDOMNode()` should have been used instead. For function components, the "instance" is an internal implementation detail of React / Preact which shouldn't be accessed externally, so we should remove this type.

Reply:
I removed instance, and that made the generics even simpler, as now we only have generics for props and state.